perma_web/perma/models.py

```diff
@@ -1109,6 +1109,64 @@ def can_view_usage_plan(self):
         return not self.nonpaying or (self.is_registrar_user() and not self.registrar.nonpaying)
 
 
+    ### merging accounts ###
+
+    def copy_memberships_from_users(self, users):
+        original_orgs = set(self.organizations.all())
+
+        orgs = set()
+        registrars = set()
+        for user in users:
+            if user.registrar_id:
+                registrars.add(user.registrar_id)
+            else:
+                orgs.update(user.organizations.all())
+
+        if orgs or registrars:
+            assert not (orgs and registrars), f"This set of users includes both org and registrar users: {', '.join([user.id for user in users])}."
+            if registrars:
+                assert len(registrars) == 1
+                new_registrar_id = registrars.pop()
+                if not self.registrar_id:
+                    self.registrar_id = new_registrar_id
+                    self.prepend_to_notes(f"Added registrar during the merging of accounts: {new_registrar_id}")
+                    self.save(update_fields=['registrar_id', 'notes'])
+            else:
+                if original_orgs != orgs:
+                    self.prepend_to_notes(f"Added organizations during the merging of accounts: {', '.join([str(o.id) for o in orgs - original_orgs])}")
+                    self.save(update_fields=['notes'])
+                    self.organizations.add(*orgs)
+
+    def soft_delete_after_merge_with_user(self, user):
```

Comment:
Does this want to be a model method? I am doing the reverse inline in the task... and using bulk operations for increased efficiency. Given that we will only do this once (🤞) and it only takes seconds, I think not worth giving it any thought but.....

Might as well keep it: easier to reference later when making a generalized soft-delete method.

Reply:
Nah.